src/escape.rs

```diff
@@ -31,6 +31,13 @@ pub struct EscapePass;
 /// ```
 declare_lint!(pub BOXED_LOCAL, Warn, "using Box<T> where unnecessary");
 
+fn is_box(ty: ty::Ty) -> bool {
```

Comment:
Since this is only used in the `true` case, an `if let ty::TyBox(..) {` won't actually be too bad.